### PR TITLE
Add with-all flag to setup.

### PR DESCRIPTION
**Pull request by PiotrMrozik**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,8 @@ set(RGL_BUILD_TESTS ON CACHE BOOL
 set(RGL_BUILD_TOOLS ON CACHE BOOL "Enables building RGL executable tools")
 
 # Extensions configuration
+set(RGL_BUILD_ALL OFF CACHE BOOL
+        "Enables all possible extensions, tools and tests.")
```
Comment (msz-rai) [inline CMakeLists.txt:30]: I'd prefer not to pollute CMakeLists with the flag `all`. We can manage it in the `setup.py` script.

File: setup.py [hunk 2]
```diff
@@ -118,6 +120,7 @@ def is_cuda_version_ok():
     # Build
     cmake_args = [
         f"-DCMAKE_TOOLCHAIN_FILE={os.path.join(cfg.VCPKG_INSTALL_DIR, 'scripts', 'buildsystems', 'vcpkg.cmake') if args.with_pcl else ''}",
+        f"-DRGL_BUILD_ALL={'ON' if args.with_all else 'OFF'}",
         f"-DRGL_BUILD_PCL_EXTENSION={'ON' if args.with_pcl else 'OFF'}",
         f"-DRGL_BUILD_ROS2_EXTENSION={'ON' if args.with_ros2 else 'OFF'}",
         f"-DRGL_BUILD_ROS2_EXTENSION_STANDALONE={'ON' if args.with_ros2_standalone else 'OFF'}"
```
Comment (msz-rai) [inline setup.py:126]:
```suggestion
        f"-DRGL_BUILD_PCL_EXTENSION={'ON' if args.with_pcl or args.with_all else 'OFF'}",
        f"-DRGL_BUILD_ROS2_EXTENSION={'ON' if args.with_ros2 or args.with_all else 'OFF'}",
        f"-DRGL_BUILD_ROS2_EXTENSION_STANDALONE={'ON' if args.with_ros2_standalone or args.with_all else 'OFF'}"
```
Suggestion to the previous comment.

Comment (msz-rai) [reply setup.py:126]: I have realized that there is no flag to build tests and tools, but I see no obstacles to adding them (`--with-tools`, `--with-tests`). What do you think?

**Review by msz-rai — CHANGES_REQUESTED** [review]
The flag you added will be handy. Please consider my suggestions.

Also, in your current implementation, `all` doesn't include ros2 extension with a standalone build.

File: setup.py (hunk 2 above)
Comment (prybicki) [reply setup.py:126]: Hmm, how about:?
```python
if args.with_all:
    args.with_pcl = True
    args.with_ros2 = True
```

Comment (PiotrMrozik) [reply setup.py:126]: It much more clear then banch of ORs.

File: CMakeLists.txt (hunk 1 above)
Comment (PiotrMrozik) [reply CMakeLists.txt:30]: I am aware of ros2standalone build.
This should be discussed. IMHO in "build all" mode we should choose "default" ros2 building method

File: setup.py (hunk 2 above)
Comment (prybicki) [reply setup.py:126]: @msz-rai Same applies to autotape, logging level, etc. These are indirectly available through `--cmake "-D...`. Not sure if it makes sense to lay pipes through setup to every option in cmake. I'd settle down only for the most important ones from user's point of view.

File: CMakeLists.txt (hunk 1 above)
Comment (msz-rai) [reply CMakeLists.txt]: @prybicki, what are your thoughts?

File: CMakeLists.txt [hunk 3]
```diff
@@ -99,7 +99,7 @@ add_library(RobotecGPULidar SHARED
 
 set_property(TARGET RobotecGPULidar PROPERTY POSITION_INDEPENDENT_CODE ON)
 
-if (RGL_BUILD_PCL_EXTENSION)
+if (RGL_BUILD_PCL_EXTENSION OR RGL_BUILD_ALL)
```
Comment (msz-rai) [inline CMakeLists.txt:102]: Leftovers after review changes. Please scan the whole CMakeLists.txt file.

File: setup.py [hunk 4]
```diff
@@ -115,6 +122,7 @@ def is_cuda_version_ok():
     if on_windows():
         os.environ["Path"] = os.environ["Path"] + ";" + os.path.join(os.getcwd(), args.build_dir)
 
+
```
Comment (msz-rai) [inline setup.py:125]:
```suggestion
```

File: setup.py [hunk 5]
```diff
@@ -44,6 +44,8 @@ def main():
                         help="Install dependencies for PCL extension and exit")
     parser.add_argument("--clean-build", action='store_true',
                         help="Remove build directory before cmake")
+    parser.add_argument("--with-all", action='store_true',
+                        help="Build RGL with all possible extensions, tools and tests.")
```
Comment (msz-rai) [inline setup.py:48]: At the moment, only extensions are handled with this flag. Add flags to enable building tools and tests. Or modify the help message.

**Review by msz-rai — CHANGES_REQUESTED** [review]

**Comment by prybicki** [comment]
At first sight, the suggestion seemed nice, however, it introduces ambiguities about what `all` means, e.g.,
- does it include ros2 standalone build?
- does it include tests and tools?
This flag can be replaced by typing individually desired `all` configuration and using reverse search in bash.

**Pull request closed without merging**